utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -15,67 +15,189 @@
+//!
+//! 1. https://substrate.dev/docs/en/knowledgebase/integrate/try-runtime
+//! 2. https://www.crowdcast.io/e/substrate-seminar/41
+//! 3. https://substrate.dev/docs/en/knowledgebase/advanced/executor
+//!
+//! ---
+//!
+//! ## Overview
+//!
+//! The basis of all try-runtime commands is the same: connect to a live node, scrape its *state*
+//! and put it inside a `TestExternalities`, then call into a *specific runtime-api* using the given
+//! state and some *runtime*.
+//!
+//! All of the variables in the above statement are made *italic*. Let's look at each of them:
+//!
+//! 1. **State** is the key-value pairs of data that consist the important information that any
+//! blockchain is keeping. A state can be full (all key-value pairs), or be partial (only pairs
+//! related to some pallets). Moreover, some keys are special and live outside of pallets, known as
+//! [`well_known_keys`] in substrate. The most important of these is the `:CODE:` key, which
+//! contains the code used for execution, when `--execution Wasm` is used.
+//! 2. *A runtime-api* is a call into the function defined in the runtime, *on top of a given
+//! state*. Each subcommand of `try-runtime` utilizes a specific one of these.
+//! 3. Finally, the **runtime** is the actual code that is used to execute the aforementioned
+//! runtime-api. All substrate based chains always have two runtimes: native and wasm. The decision
+//! of which one is chosen is slightly non-trivial. First, let's look at the options:
+//!
+//! 1. Native: this means that the runtime that is **in your codebase**, aka whatever you see in
+//! your editor is being used. This runtime is easier for diagnostics. In the rest of the
+//! documentation, by "local runtime", we mean the na 2. Wasm: this means that whatever is stored in
+//! the `:CODE:` key of the state that your scrape is being used. In plain sight, since the entire
+//! state (including `:CODE:`) is scraped from a remote chain, you could conclude that the wasm
+//! runtime, if used, is always equal to the canonical runtime. That's factually true, but then the
+//! testing would be quite lame. Typically, with try-runtime, you don't want to execute whatever
+//! code is already on the live chain. Instead, you want your local runtime (which typically
+//! includes a non-released feature) to be used. This is why try-runtime overwrites the wasm runtime
+//! (at `:CODE:`) with the local runtime as well. That being said, this behavior can be controlled
+//! in certain subcommands with a special flag.
+//!
+//! The decision of which runtime is eventually used is based on two facts:
+//!
+//! 1. `--execution` flag. If you specify `wasm`, then it is *always* wasm. If it is `native`, then
+//! if and ONLY IF the spec versions match, then the native runtime is used. Else, wasm runtime is
+//! used again.
+//! 2. `--chain` flag (if present in your cli), which determines *which native runtime*, if needed,
+//! might be used.
```

Comment:
This only affects native? What if you are in polkadot repo and want to use westend runtime with wasm, how would specify that?

Reply:
you can do that with `--chain westend-dev --execution Wasm --overwrite-code`. I made the doc a bit clearer, let me know if it is better now.